File: Core/OrderAggregation/Order.cs
using Core.Entities;

namespace Core.OrderAggregation
{
    public class Order : SuperEntity
    {
        public Order()
        {
        }

        public Order(IReadOnlyList<OrderItem> orderItems, string userEmail, Address shippingAddress, DelieveryMethod delieveryMethod, decimal subTotal)
        {
            UserEmail = userEmail;
            ShippingAddress = shippingAddress;
            DelieveryMethod = delieveryMethod;
            OrderItems = orderItems;
            SubTotal = subTotal;
        }

        public string UserEmail { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public Address ShippingAddress { get; set; }
        public DelieveryMethod DelieveryMethod { get; set; } 
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public decimal SubTotal { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public string PaymentIntentId { get; set; }

        public decimal GetTotal() {
            return SubTotal + DelieveryMethod.PriceOfDelievery;
        }
    }
}

File: client/src/app/shared/models/order.ts
import { Address } from "./user"

export interface OrderToCreate {
    basketId: string;
    delieveryMethodId: number;
    shippingAddress: Address;
}

export interface Order {
    id: number
    userEmail: string
    orderDate: string
    shippingAddress: Address
    delieveryMethod: string
    shippingPrice: number
    orderItems: OrderItem[]
    subTotal: number
    total: number
    orderStatus: string
  }
  
export interface OrderItem {
    id: number
    name: string
    pictureUrl: string
    price: number
    quantity: number
  }
  